main.cpp

```diff
@@ -475,7 +475,10 @@ int main(int argc, char ** argv) {
                 std::string line;
                 bool another_line = true;
                 do {
-                    std::getline(std::cin, line);
+                    if (!std::getline(std::cin, line)) {
+                        // input stream is bad or EOF received
+                        return 0;
```

Comment:
Did you mean 'break'?

Reply:
With `break`, we'd be repeating the outer loop which will immediately ask for a prompt again. Since CTRL-C just calls exit, I thought "return" would be a reasonable choice. There's nothing interesting happening in the rest of `main()` anyway ;)

Reply:
> Since CTRL-C just calls `exit`, I thought `return` would be a reasonable choice. There's nothing interesting happening in the rest of `main()` anyway ;)

It actually does not though, when interacting it sets "is_interacting = false" , breaking the loop and returning control to the bot. So if the aim is to replicate functionality of CTRL+C , then you're not doing that. But if you want a terminate/kill button, that works. Tbh a terminate/kill button would be good to gtfo faster than with CTRL+C.

https://github.com/ggerganov/llama.cpp/blob/d7ed3a4a4a6bc61007e9c5c328725a57efc664e2/main.cpp#L143-L155

Reply:
Agree, let's keep it as terminate/kill since that's what EOF is generally used for